com.musala.simple.students.db/src/main/java/com/musala/simple/students/db/MyMongoDatabase.java

```diff
@@ -0,0 +1,258 @@
+package com.musala.simple.students.db;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoWriteException;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.IndexOptions;
+
+import java.lang.reflect.Field;
+
+import com.musala.simple.students.db.exception.StudentNotFoundException;
+import com.musala.simple.students.db.internal.ErrorMessage;
+import com.musala.simple.students.db.internal.InfoMessage;
+import com.musala.simple.students.db.student.Student;
+
+/**
+ * MongoDb implementation of the {@link Database} interface.
+ * 
+ * @author yoan.petrushinov
+ *
+ */
+public class MyMongoDatabase implements Database {
+	private MongoClient mongo;
+	private MongoDatabase database;
+	private MongoCollection<Document> studentsCollection;
+	private String dbName;
+	private Logger logger = LoggerFactory.getLogger(Main.class);
```

Comment:
Every time you declare a logger, it has to be bound to the current class (this.getClass()). In such way, logs will be printed as being emitted by this class. If you leave this logger with `getLogger(Main.class)`, logs from `MyMongoDatabase` will be listed as logged by the `Main` which is incorrect. Please correct this everywhere.